src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -108,38 +108,45 @@ public void Start()
             stopping = false;
             pipelineInvocationTasks = new ConcurrentDictionary<Task, Task>();
 
-            
-            for(var i = 0; i < numberOfClients; i++)
+            var exceptions = new ConcurrentQueue<Exception>();
+            Parallel.For(0, numberOfClients, i =>
```

Comment:
@yvesgoeleven even then its not possible as `CircularBuffer` isn't async. Maybe the only thing that works is to make the circular buffer contain Lazy items of Task<IMessageReceiver>. So something like:

`CircularBuffer<Lazy<Task<IMessageReceiver>>>`

That would resolve the item async on first access but I'm not sure if that even works :-) @danielmarbach any guidance? Does my idea make sense?